debezium-core/src/main/java/io/debezium/relational/topic/ByDatabasePlatformTopicMapper.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.relational.topic;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author David Leibovic
+ * @see this doc for useful terminology definitions: https://docs.google.com/document/d/12KZDiv5uUvlB8jnJQJ8KK7PR6whPoWUQEWpX-Ly8n3A/edit#
+ */
+public class ByDatabasePlatformTopicMapper extends TopicMapper {
```

Comment:
I will think about if a generic version is possible. I think we'd definitely need some config options available for `TopicMapper` implementations to use. Should those config options be generic, i.e. `topic.mapper.config1` and `topic.mapper.config2`, such that any `TopicMapper` implementation can use the configs for anything they want?

Or would it make sense to have specific config options for the "out of the box" `TopicMapper` implementations that are provided, i.e. `topic.mapper.sharding.regex` ?